src/key-value-pairs/__tests__/key-value-pairs.test.tsx

```diff
@@ -0,0 +1,136 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import * as React from 'react';
+import { render } from '@testing-library/react';
+import createWrapper from '../../../lib/components/test-utils/dom';
+import KeyValuePairs, { KeyValuePairsProps } from '../../../lib/components/key-value-pairs';
+
+function renderKeyValuePairs(props: KeyValuePairsProps) {
+  const { container, debug } = render(<KeyValuePairs {...props} />);
+  return { wrapper: createWrapper(container).findKeyValuePairs()!, debug };
+}
+
+describe('KeyValuePairs', () => {
+  describe('columns', () => {
+    test(`renders 1 column correctly`, () => {
+      const { wrapper } = renderKeyValuePairs({
+        columns: [
+          {
+            items: [
+              {
+                label: 'Label for key',
+                value: 'Value',
+              },
+            ],
+          },
+        ],
+      });
+
+      expect(wrapper.findColumns()).toHaveLength(1);
+    });
+
+    test(`renders multiple columns correctly`, () => {
+      const { wrapper } = renderKeyValuePairs({
+        columns: [
+          {
+            items: [
+              {
+                label: 'Label for key',
+                value: 'Value',
+              },
+            ],
+          },
+          {
+            items: [
+              {
+                label: 'Label for key',
+                value: 'Value',
+              },
+            ],
+          },
+        ],
+      });
+
+      expect(wrapper.findColumns()).toHaveLength(2);
+    });
+  });
+
+  test('renders title correctly', () => {
+    const { wrapper } = renderKeyValuePairs({
+      columns: [
+        {
+          title: 'Title',
+          items: [
+            {
+              label: 'Label for key',
+              value: 'Value',
+            },
+          ],
+        },
+      ],
+    });
+
+    expect(wrapper.findColumns()[0].findTitle()!.getElement()).toHaveTextContent('Title');
+  });
+
+  test('renders pairs correctly', () => {
+    const { wrapper } = renderKeyValuePairs({
+      columns: [
+        {
+          title: 'Column title 1',
+          items: [
+            {
+              label: 'Label for key 1',
+              value: 'Value 1',
+            },
+            {
+              label: 'Label for key 2',
+              value: 'Value 2',
+            },
+          ],
+        },
+        {
+          title: 'Column title 2',
+          items: [
+            {
+              label: 'Label for key 3',
+              value: 'Value 3',
+            },
+            {
+              label: 'Label for key 4',
+              value: 'Value 4',
+            },
+          ],
+        },
+      ],
+    });
+
+    expect(wrapper.findColumns()[0].findTitle()!.getElement().textContent).toBe('Column title 1');
```

Comment:
We have `toHaveTextContent` matcher

Reply:
`toHaveTextContent` checks for text entry, while `textContent` along with `toBe` checks the exact string matching

Reply:
Jest-dom has a guidance for this case: https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohavetextcontent

They recommend using regexp for it

```js
expect(element).toHaveTextContent(/^Text Content$/) // to match the whole content
```

Feel free to apply their suggestion or keep your way, no much difference for me